packages/core/core/src/PackagerRunner.js

```diff
@@ -43,37 +49,49 @@ export default class PackagerRunner {
     this.pluginOptions = new PluginOptions(this.options);
   }
 
-  async writeBundle(bundle: InternalBundle, bundleGraph: InternalBundleGraph) {
-    let {inputFS, outputFS} = this.options;
-    let start = Date.now();
-
+  async getBundleResult(
+    bundle: InternalBundle,
+    bundleGraph: InternalBundleGraph
+  ): Promise<{|contents: Blob, map: Readable | string | null|}> {
     let result, cacheKey;
     if (!this.options.disableCache) {
       cacheKey = await this.getCacheKey(bundle, bundleGraph);
-      result = await this.readFromCache(cacheKey);
-    }
+      let cacheResult: ?{|
+        contents: Readable,
+        map: ?Readable
+      |} = await this.readFromCache(cacheKey);
 
-    if (!result) {
-      let packaged = await this.package(bundle, bundleGraph);
-      let res = await this.optimize(
-        bundle,
-        bundleGraph,
-        packaged.contents,
-        packaged.map
-      );
-
-      let map = res.map ? await this.generateSourceMap(bundle, res.map) : null;
-      result = {
-        contents: res.contents,
-        map
-      };
-
-      if (cacheKey != null) {
-        await this.writeToCache(cacheKey, result.contents, map);
+      if (cacheResult) {
+        result = cacheResult;
       }
     }
 
-    let {contents, map} = result;
+    let packaged = await this.package(bundle, bundleGraph);
+    let res = await this.optimize(
+      bundle,
+      bundleGraph,
+      packaged.contents,
+      packaged.map
+    );
+
+    let map = res.map ? await this.generateSourceMap(bundle, res.map) : null;
```

Comment:
Perhaps we should force inline source maps for inline bundles? Looks like right now the source map just gets thrown away by the HTMLPackager, but if it were generated inline it could work. I guess we don't support inline source maps at all right now, so we could add that later.

Reply:
My thought was that we would add inline later once we have support for it. 

How best can we track that? Create an issue? Throw in a `TODO`

Reply:
An issue would be great.

Reply:
I created #3490, feel free to beef it up with any more context that I may be missing.